toolchain/check/testdata/interop/cpp/class/access.carbon

```diff
@@ -529,6 +529,204 @@ class D {
+}
+
+// --- fail_import_overload_set_private_base_class.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "overload_set_base_class.h";
+
+fn F() {
+  // CHECK:STDERR: fail_import_overload_set_private_base_class.carbon:[[@LINE+5]]:3: error: cannot access private member `foo` of type `Cpp.PrivateDerived` [ClassInvalidMemberAccess]
+  // CHECK:STDERR:   Cpp.PrivateDerived.foo();
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_import_overload_set_private_base_class.carbon: note: declared here [ClassMemberDeclaration]
+  // CHECK:STDERR:
+  Cpp.PrivateDerived.foo();
+  // CHECK:STDERR: fail_import_overload_set_private_base_class.carbon:[[@LINE+5]]:3: error: cannot access private member `foo` of type `Cpp.PrivateDerived` [ClassInvalidMemberAccess]
+  // CHECK:STDERR:   Cpp.PrivateDerived.foo(1);
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_import_overload_set_private_base_class.carbon: note: declared here [ClassMemberDeclaration]
+  // CHECK:STDERR:
+  Cpp.PrivateDerived.foo(1);
+  // CHECK:STDERR: fail_import_overload_set_private_base_class.carbon:[[@LINE+5]]:3: error: cannot access private member `foo` of type `Cpp.PrivateDerived` [ClassInvalidMemberAccess]
+  // CHECK:STDERR:   Cpp.PrivateDerived.foo(1, 2);
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR: fail_import_overload_set_private_base_class.carbon: note: declared here [ClassMemberDeclaration]
+  // CHECK:STDERR:
+  Cpp.PrivateDerived.foo(1, 2);
+}
+
+// --- fail_import_overload_set_private_base_class_derived.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "overload_set_base_class.h";
+
+class Derived {
+  extend base: Cpp.PrivateDerived;
+
+  fn F() {
+    // CHECK:STDERR: fail_import_overload_set_private_base_class_derived.carbon:[[@LINE+5]]:5: error: cannot access private member `foo` of type `Cpp.PrivateDerived` [ClassInvalidMemberAccess]
+    // CHECK:STDERR:     foo();
+    // CHECK:STDERR:     ^~~~~
+    // CHECK:STDERR: fail_import_overload_set_private_base_class_derived.carbon: note: declared here [ClassMemberDeclaration]
+    // CHECK:STDERR:
+    foo();
+    // CHECK:STDERR: fail_import_overload_set_private_base_class_derived.carbon:[[@LINE+5]]:5: error: cannot access private member `foo` of type `Cpp.PrivateDerived` [ClassInvalidMemberAccess]
+    // CHECK:STDERR:     foo(1);
+    // CHECK:STDERR:     ^~~~~~
+    // CHECK:STDERR: fail_import_overload_set_private_base_class_derived.carbon: note: declared here [ClassMemberDeclaration]
+    // CHECK:STDERR:
+    foo(1);
+    // CHECK:STDERR: fail_import_overload_set_private_base_class_derived.carbon:[[@LINE+5]]:5: error: cannot access private member `foo` of type `Cpp.PrivateDerived` [ClassInvalidMemberAccess]
+    // CHECK:STDERR:     foo(1, 2);
+    // CHECK:STDERR:     ^~~~~~~~~
+    // CHECK:STDERR: fail_import_overload_set_private_base_class_derived.carbon: note: declared here [ClassMemberDeclaration]
+    // CHECK:STDERR:
+    foo(1, 2);
+  }
+}
+
```

Comment:
Can you add a test that does just a little more access control layering, such as:

```suggestion

// --- layered_inheritance_test.carbon

library "[[@TEST_NAME]]";

import Cpp inline '''
class C {
  public:
    static auto Public() -> void;
  protected:
    static auto Protected() -> void;
  private:
    static auto Private() -> void;
};

class D : private C {};

class E : public D {};
''';

class F {
  extend base: Cpp.E;

  fn G[self: Self]() {
    Self.Public();
    Self.Protected();
    Self.Private();
  }
}
```

Also, are non-static methods intended to work? If so, perhaps it'd be good to test those too?